store/commitment/store.go

```diff
@@ -126,13 +227,26 @@ func (c *CommitStore) SetInitialVersion(version uint64) error {
 	return nil
 }
 
-func (c *CommitStore) GetProof(storeKey string, version uint64, key []byte) (*ics23.CommitmentProof, error) {
+func (c *CommitStore) GetProof(storeKey string, version uint64, key []byte) (cmtcrypto.ProofOps, error) {
 	tree, ok := c.multiTrees[storeKey]
 	if !ok {
-		return nil, fmt.Errorf("store %s not found", storeKey)
+		return cmtcrypto.ProofOps{}, fmt.Errorf("store %s not found", storeKey)
+	}
+
+	proof, err := tree.GetProof(version, key)
+	if err != nil {
+		return cmtcrypto.ProofOps{}, err
+	}
+	cInfo, err := c.GetCommitInfo(version)
+	if err != nil {
+		return cmtcrypto.ProofOps{}, err
+	}
+	if cInfo == nil {
+		return cmtcrypto.ProofOps{}, fmt.Errorf("commit info not found for version %d", version)
 	}
+	commitOp := store.NewIAVLCommitmentOp(key, proof).ProofOp()
 
-	return tree.GetProof(version, key)
+	return cmtcrypto.ProofOps{Ops: []cmtcrypto.ProofOp{commitOp, cInfo.ProofOp(storeKey)}}, nil
 }
 
 func (c *CommitStore) Prune(version uint64) (ferr error) {
```

Comment:
do you think we need to prune the `CommitInfo` metadata here?

Reply:
Yes 👍